lib/models/introductory_price.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'introductory_price.freezed.dart';
+part 'introductory_price.g.dart';
+
+@freezed
+
+/// Contains all the introductory information associated with a [Product]
+class IntroductoryPrice with _$IntroductoryPrice {
+  const factory IntroductoryPrice(
+    /// Introductory price of a subscription in the local currency.
+    @JsonKey(name: 'price') double introPrice,
```

Comment:
Could you help me with this test pls? I think that response is not valid

![image](https://user-images.githubusercontent.com/20377394/142277980-e858442b-5eec-492e-8f9d-bf34af2d24e8.png)

Reply:
The response is fine. The issue is that the JsonKey to use in product_wrapper for the `introductoryPrice` should be `introPrice` instead of `intro_price`. `introductory_price.dart` also needs to be updated to match the keys in the `introPrice` object of the response:

```
'introPrice': {
          'price': 0,
          'priceString': '\$0.00',
          'period': 'P1W',
          'cycles': 1,
          'periodUnit': 'DAY',
          'periodNumberOfUnits': 7
  },
 ```
 
The information is duplicated in the response, because we used to have the intro price information in the root of the JSON, then we moved it to its own object `introPrice` and changed the json keys to be camelcase. We left the keys in the JSON root because other SDKs still use those :)
 
 Let me know if you want me to do the update to fix the test :)